v2/api/dataprotection/customizations/backup_vaults_backup_instance_extensions.go

```diff
@@ -19,14 +19,26 @@ import (
 	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
 	. "github.com/Azure/azure-service-operator/v2/internal/logging"
 	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/internal/set"
 	"github.com/Azure/azure-service-operator/v2/internal/util/to"
 	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
 	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
 )
 
 var _ extensions.PostReconciliationChecker = &BackupVaultsBackupInstanceExtension{}
 
-var protectionError = "ProtectionError"
+var terminalStates = set.Make(
```

Comment:
Can you add some documentation about how we determined this list, and what exactly it means to be terminal here?

Reply:
These are the states listed and we do no foresee any frequent changed here as these states are not usually changed.
 
Swagger: https://github.com/Azure/azure-rest-api-specs/blob/a651ba25cda4eec698a3a4e35f867ecc2681d126/specification/dataprotection/resource-manager/Microsoft.DataProtection/stable/2023-11-01/dataprotection.json#L5014

Public Doc: (might not be updated with latest swagger)
https://learn.microsoft.com/en-us/rest/api/dataprotection/backup-instances/get?view=rest-dataprotection-2024-04-01&tabs=HTTP#status

Reply:
Adding these URLs as comments would be useful, so that a future maintainer can go verify if things change in the future.

Reply:
Done!